Assets/Scripts/Interceptors/InterceptorBase.cs

```diff
@@ -115,8 +155,8 @@ private void OnTriggerEnter(Collider other) {
       float killProbability = threat.StaticConfig.HitConfig?.KillProbability ?? 1;
       bool isHit = Random.value <= killProbability;
       if (isHit) {
-        OnHit?.Invoke(this);
         threat.HandleIntercept();
+        OnHit?.Invoke(this);
```

Comment:
Seems like order of operations matters here, could you add a comment explaining why you ordered it this way?

Reply:
I forgot lol.  Looking at where `OnHit` event is used, it doesn't seem like it matters.